### PR TITLE
Version: Handle if git is not installed

### PR DESCRIPTION
**Pull request by christianmemije**
## Summary

* Attempts to handle case in which git is not installed.
* Meant to fix 1/2 of https://github.com/learningequality/kolibri-kivy/issues/11

## TODO

- [ ] Write a test

## Reviewer guidance

* How to write a test for this?

### CODE REVIEW

File: kolibri/utils/version.py [hunk 1]
```diff
@@ -228,19 +228,21 @@ def get_git_describe():
...
+            stderr=subprocess.PIPE,
+            shell=True,
+            cwd=repo_dir,
+            universal_newlines=True
+        )
+        # This does not fail if git is not available or current dir isn't a git
+        # repo - it's safe.
+        version_string = p.communicate()[0].rstrip()
+        return version_string
+    except:
```
Comment (rtibbles) [inline kolibri/utils/version.py]: Should avoid blanket `except` clauses. Catch the exception you are trying to catch.

If you are catching multiple exceptions `except (ValueError, IOError):`

Comment (christianmemije) [reply kolibri/utils/version.py]: @rtibbles Updated!

**Comment by rtibbles** [comment]
I think this probably doesn't need a test, because to do it you would end up mocking the `subprocess.Popen`, to throw the errors you are trying to catch, which is then just testing that Python can catch exceptions properly.

The better test would be to have an environment that does not have git and check that we can still run the version code.

If you want to make sure this change does not get reverted, the simple mock test would be sufficient.

My personal preference would be to use mock patch to do this though, so for the new test you create within the above TestCase:
```
@patch('kolibri.utils.version.subprocess.Popen', side_effect=EnvironmentError):
def test_no_git_raise_caught(self):
        v = get_git_version()
        self.assertEqual(None, v)
```

and add the import for patch: `from mock import patch`

**Comment by benjaoming** [comment]
Hi @christianmemije - thanks for looking into this.

I just want to add that *I did test this without git*, so in case this is based on suspicion, your issue might not be fixed.

For instance, I can run a command `none-existing-cmd-name` and it will not raise `EnvironmentError`:

```
import subprocess
repo_dir="/tmp"
git_log = subprocess.Popen(
    'none-existing-cmd-name log --pretty=format:%ct --quiet -1 HEAD',
    stdout=subprocess.PIPE,
    stderr=subprocess.PIPE,
    shell=True,
    cwd=repo_dir,
    universal_newlines=True
)
# This does not fail if git is not available or current dir isn't a git
# repo - it's safe.
timestamp = git_log.communicate()[0]
```

**Comment by christianmemije** [comment]
Hey @benjaoming. After further debugging, I realized that the actual issue is that in android, the shell executable is in `/system/bin/sh` not in `/bin/sh`, so the Popen function cannot execute the proper shell to even run a command: Adding `executable="/system/bin/sh"` as an argument for Android fixes this. A fix was added to the codebase, but that is not in 2.7.

Issue: https://bugs.python.org/issue16255 
Fix: https://hg.python.org/cpython/rev/96a9992d1003

So this is actually what is being caught, not a lack of git, as you mentioned.

**Pull request merged**